src/opendr/perception/object_detection_2d/yolov5/yolov5_learner.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2020-2022 OpenDR European Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# General imports
+import os
+import time
+import json
+import numpy as np
+import warnings
+from tqdm import tqdm
+from urllib.request import urlretrieve
+
+# OpenDR engine imports
+from opendr.engine.learners import Learner
+from opendr.engine.data import Image
+from opendr.engine.target import BoundingBox, BoundingBoxList
+from opendr.engine.constants import OPENDR_SERVER_URL
+
+# yolov5 imports
+import torch
+
+
+class YOLOv5DetectorLearner(Learner):
+    available_models = ['yolov5s', 'yolov5n', 'yolov5m', 'yolov5l', 'yolov5x',
+                        'yolov5n6', 'yolov5s6', 'yolov5m6', 'yolov5l6', 'custom']
+
+    def __init__(self, model_name, path=None, device='cuda', temp_path='.'):
+        super(YOLOv5DetectorLearner, self).__init__(device=device, temp_path=temp_path)
+        if model_name not in self.available_models:
+            model_name = 'yolov5s'
+            print('Unrecognized model name, defaulting to "yolov5s"')
+
+        default_dir = torch.hub.get_dir()
+        torch.hub.set_dir(temp_path)
+
+        if path is None:
+            self.model = torch.hub.load('ultralytics/yolov5:master', 'custom', f'{temp_path}/{model_name}',
+                                        force_reload=True)
```

Comment:
Should we change this to `False`, to avoid re-downloading the model every time?

Reply:
Added as a parameter for the user, as I found I had some caching issues and needed to set it to True but I agree it shouldn't be True by default